5_databases/week1/hyf_homework.sql

```diff
@@ -0,0 +1,62 @@
+-- Active: 1698992883180@@127.0.0.1@3306@hyf_homework1
+--1.Find out how many tasks are in the task table--
+SELECT COUNT (*)
+FROM task;
+
+--2.Find out how many tasks in the task table do not have a valid due date
+SELECT COUNT(*)
+FROM task
+WHERE due_date IS NULL;
+
+
+--3.Find all the tasks that are marked as done--
+
+SELECT*FROM task
+WHERE status_id = 3;
+
+--4.Find all the tasks that are not marked as done
+
+SELECT*FROM task
+WHERE status_id != 3;
+
+--5.Get all the tasks, sorted with the most recently created first
+
+SELECT * FROM task
+ORDER BY created DESC;
+
+--6.Get the single most recently created task
+
+SELECT * FROM task
+ORDER BY created DESC
+LIMIT 1;
+
+--7.Get the title and due date of all tasks where the title or description contains database
+
+SELECT title, due_date
+FROM task
+WHERE title LIKE '%database%' OR description LIKE '%database%';
+
+--8.Get the title and status (as text) of all tasks
+
+SELECT task.title,status.name AS task_status
```

Comment:
Good job! This needs a small tweak tho. This returns the task 3 times with all statuses which means 105 rows of tasks, but we only have 35 tasks. We only want 1 task and its current status:

Change the last line and analyse what is different from what you wrote:
`ON status.id=task.status_id;`

Reply:
Corrected.Thank you for the review.